src/components/MembershipsList/index.js

```diff
@@ -0,0 +1,127 @@
+import React, { useState, useEffect } from 'react';
+import { Button, Card, CardContent, CardHeader, Grid, List, Typography } from '@material-ui/core';
+import { Link } from 'react-router-dom';
+import activity from '../../services/activity';
+import membershipService from '../../services/membership';
+import userService from '../../services/user';
+import GordonSnackbar from '../Snackbar';
+import GordonLoader from '../Loader';
+import MembershipInfoCard from './components/MembershipInfoCard';
+import './index.css';
+
+/**
+ * A List of memberships for display on the Profile and MyProfile views.
+ * @param {string} user Either the user's ID number for MyProfile or the username for Profile
+ * @param {boolean} myProf Whether this is shown in MyProfile or not
+ * @returns {JSX} A list of the user's memberships
+ */
+const MembershipsList = ({ user, myProf }) => {
+  const [memberships, setMemberships] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [snackbar, setSnackbar] = useState({ message: '', severity: '', open: false });
+
+  useEffect(() => {
+    async function loadMemberships() {
+      setLoading(true);
+      if (myProf) {
+        const myMemberships = await userService.getMembershipsAlphabetically(user);
+        await Promise.all(
+          myMemberships.map(async (membership) => {
+            const involvement = await activity.get(membership.ActivityCode);
+            membership.IsInvolvementPrivate = involvement.Privacy;
+          }),
+        );
+        setMemberships(myMemberships);
+      } else {
+        const publicMemberships = await userService.getPublicMemberships(user);
+        setMemberships(publicMemberships);
+      }
+      setLoading(false);
+    }
+    loadMemberships();
+  }, [myProf, user]);
+
+  const MembershipsList = () => {
+    if (memberships.length === 0) {
```

Comment:
Wow, nice simplification here by removing the redundant code that gets public memberships.